sql/as_of_test.go

```diff
@@ -178,19 +192,40 @@ func TestAsOfRetry(t *testing.T) {
 	const val1 = 1
 	const val2 = 2
 	const name = "boulanger"
+	var cls int64
+	var clss string
 
-	if _, err := sqlDB.Exec(fmt.Sprintf(`
+	if err := sqlDB.QueryRow(fmt.Sprintf(`
 			CREATE DATABASE d;
 			CREATE TABLE d.t (s STRING PRIMARY KEY, a INT);
-			INSERT INTO d.t (s, a) VALUES ('%v', %v);
-		`, name, val1)); err != nil {
+			INSERT INTO d.t (s, a) VALUES ('%v', %v)
+			RETURNING cluster_logical_timestamp()::int, cluster_logical_timestamp()::string;
+		`, name, val1)).Scan(&cls, &clss); err != nil {
 		t.Fatal(err)
 	}
+	t.Logf("CLS 1: %v", cls)
+	t.Logf("CLSS 1: %v", clss)
+	tsStart := time.Unix(0, cls+1).Format(time.RFC3339Nano)
+
+	// Wait until now() is after tsStart so that we can subtract 1 from tsVal1
+	// later on. This still isn't totally safe, but it prevents test flakiness.
+	util.SucceedsSoon(t, func() error {
+		var afterNow bool
+		if err := sqlDB.QueryRow("SELECT now() >= $1", tsStart).Scan(&afterNow); err != nil {
+			return err
+		}
+		if !afterNow {
+			return errors.New("timestamp not after now()")
+		}
+		return nil
+	})
```

Comment:
I suppose you are calling this twice, so you might as well write a function that implements this.